### PR TITLE
HTBHF-2534 Removed unused columns, added dob, removed H2 db in test, upgraded spring

### PR DESCRIPTION
**Pull request by YetAnotherMatt**
When deleting the columns I left earnings_threshold_exceeded as it might be a useful flag we can use to decide eligibility status.

### CODE REVIEW

File: api/build.gradle [hunk 1]
```diff
@@ -1,7 +1,7 @@
 buildscript {
     ext {
         junitVersion = '5.3.2'
-        jacksonDataformatVersion = '2.9.10'
+        jacksonDataformatVersion = '2.10.0'
```
Comment (mikeSmith08) [inline api/build.gradle:4]: I'm surprised this worked without changes to SpringFox, I got a NoClassDefFoundError when trying to run it with this version, did you change anything else? I notice the swagger is different...

File: api/build.gradle [hunk 2]
```diff
@@ -71,7 +71,7 @@ dependencies {
     testImplementation "org.junit.jupiter:junit-jupiter-params:${junitVersion}"
     testImplementation "org.junit.vintage:junit-vintage-engine:${junitVersion}"
     testImplementation "org.assertj:assertj-core:3.11.1"
-    testImplementation "com.h2database:h2:1.4.197"
+    testImplementation "io.zonky.test:embedded-database-spring-test:1.4.1"
```
Comment (mikeSmith08) [inline api/build.gradle:74]: Why did you need to change this?

**Comment by mikeSmith08** [comment]
Why do you think earnings threshold exceeded will be useful when its not returned from DWP and its not currently stored on any of the entities? I haven't seen that used in any of the decisions made in the business rules so I'm a bit puzzled by that decision.

File: api/build.gradle (hunk 1 above)
Comment (YetAnotherMatt) [reply api/build.gradle:4]: I updated Spring as well

**Review by mikeSmith08 — APPROVED** [review]

**Comment by YetAnotherMatt** [comment]
> Why do you think earnings threshold exceeded will be useful when its not returned from DWP and its not currently stored on any of the entities? I haven't seen that used in any of the decisions made in the business rules so I'm a bit puzzled by that decision.

I just thought it could be a useful proxy for eligibility status. If the earnings threshold is exceeded then they're not eligible.

File: api/build.gradle (hunk 2 above)
Comment (YetAnotherMatt) [reply api/build.gradle:74]: Because I was getting an error initialising Flyway, and wanted to eliminate H2s a possible cause. Once I moved to Postgres I got a much more informative error back.

**Review by dwybourn — APPROVED** [review]

**Review by gethyn1 — APPROVED** [review]
